.travis.yml

```diff
@@ -1,18 +1,20 @@
-language: python
-before_install:
-  - sudo apt-get update
-
-  # First install libcangjie
-  - sudo apt-get install libsqlite3-dev
-  - git clone git://github.com/Cangjians/libcangjie
-  - cd libcangjie && git checkout 1.x && ./autogen.sh --prefix=/usr && make && sudo make install && cd ..
-  - git clean -dfx
-
-  # Next install pycangjie
-  - sudo apt-get install cython python3-dev
-  - git clone git://github.com/Cangjians/pycangjie
-  - cd pycangjie && git checkout 1.x && ./autogen.sh --prefix=/usr && make && sudo make install && cd ..
-  - git clean -dfx
-
-  - sudo apt-get install autopoint intltool libibus-1.0-dev python3-gobject
-script: ./autogen.sh && make && make check && make distcheck
+language: c
+sudo: required
+branches:
+  only:
+    - master
+services:
+  - docker
+env:
+  - OS=fedora-latest
+  - OS=ubuntu-devel
+  - OS=ubuntu-latest
+  - OS=ubuntu-lts
+install:
```

Comment:
In the other `.travis.yml` files we used `before_install`, but here you use `install`. Why?

Should we use `before_install` here as well? Should we use `install` in libcangjie and pycangjie?

Reply:
No particular reason here.

Conceptually, the `before_install` section is for installing prerequisites and `install` is to build and install the software for test. Since we're using docker here, the difference is not significant.

I'll change to `before_install` here.

Reply:
> Conceptually, the before_install section is for installing prerequisites and install is to build and install the software for test.

Ah, so then we should have used the following everywhere I guess?

```yaml
before_install:
  - sudo apt-get update && sudo apt-get -y install docker-ce
  - docker pull "cangjians/pycangjie:${OS}"
install:
  - docker build --build-arg OS=${OS} --tag "cangjians/ibus-cangjie:${OS}" .
```

> Since we're using docker here, the difference is not significant.

Indeed, the separation feels purely academic in our case: it would be more correct, but doesn't actually gain us anything.

> I'll change to before_install here.

Alright, thanks. 🙂